src/sentry/integrations/discord/actions/metric_alert.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import sentry_sdk
+
+from sentry import features
+from sentry.incidents.charts import build_metric_alert_chart
+from sentry.incidents.models import AlertRuleTriggerAction, Incident, IncidentStatus
+from sentry.integrations.discord.client import DiscordClient
+from sentry.integrations.discord.message_builder.metric_alerts import (
+    DiscordMetricAlertMessageBuilder,
+)
+from sentry.shared_integrations.exceptions.base import ApiError
+
+from ..utils import logger
+
+
+def send_incident_alert_notification(
+    action: AlertRuleTriggerAction,
+    incident: Incident,
+    metric_value: int,
+    new_status: IncidentStatus,
+) -> None:
+    chart_url = None
+    if features.has("organizations:metric-alert-chartcuterie", incident.organization):
+        try:
+            chart_url = build_metric_alert_chart(
+                organization=incident.organization,
+                alert_rule=incident.alert_rule,
+                selected_incident=incident,
+            )
+        except Exception as e:
+            sentry_sdk.capture_exception(e)
+
+    channel = action.target_identifier
+
+    if not channel:
+        # We can't send a message if we don't know the channel
+        logger.warning(
+            "discord.metric_alert.message_send_failure",
+            extra={"guild_id": incident.identifier, "channel_id": channel},
```

Comment:
Isn't `channel` None here? Is it valuable to add it to the logs?

Reply:
Could a better message be "Message send failure: missing channel id"?

Reply:
Do we even need to check if the channel is empty?

Reply:
I just mean since we're already checking `if not channel` and this logger message isn't used anywhere else, it's not really valuable to add the `extra` `channel_id` data containing nothing. If we were logging the same thing in another place it failed and setting the channel id there, I could see why we'd do this (to differentiate between failing with and without a channel ID). Since this isn't used anywhere else you can give it a more specific name like `discord.metric_alert.no_channel` or something.